examples/name-service/programs/name-service/src/lib.rs

```diff
@@ -76,6 +77,44 @@ pub struct NameRecord {
     #[truncate]
     pub name: String,
     pub rdata: RData,
+    #[nested]
+    pub nested: NestedData,
+}
+
+// Illustrates nested hashing feature.
+#[derive(LightHasher, Clone, Debug, AnchorSerialize, AnchorDeserialize)]
+pub struct NestedData {
```

Comment:
integration tests are in `light-protocol/test-programs`. We should make a macro test program there.